pkg/curation/metadata_test.go

```diff
@@ -10,43 +10,51 @@ import (
 
 func TestMetadataParse(t *testing.T) {
 	tests := []struct {
-		fixture  string
-		expected Metadata
-		err      bool
+		fixture       string
+		expected      *Metadata
+		expectedError string
 	}{
 		{
 			fixture: "test-fixtures/metadata-gocase",
-			expected: Metadata{
+			expected: &Metadata{
 				Built:    time.Date(2020, 06, 15, 14, 02, 36, 0, time.UTC),
 				Version:  2,
 				Checksum: "sha256:dcd6a285c839a7c65939e20c251202912f64826be68609dfc6e48df7f853ddc8",
 			},
 		},
 		{
 			fixture: "test-fixtures/metadata-edt-timezone",
-			expected: Metadata{
+			expected: &Metadata{
 				Built:    time.Date(2020, 06, 15, 18, 02, 36, 0, time.UTC),
 				Version:  2,
 				Checksum: "sha256:dcd6a285c839a7c65939e20c251202912f64826be68609dfc6e48df7f853ddc8",
 			},
 		},
+		{
+			fixture:       "/dev/null/impossible",
+			expectedError: "unable to check if DB metadata path exists (/dev/null/impossible/metadata.json): stat /dev/null/impossible/metadata.json: not a directory",
```

Comment:
nit: mind checking the existence of an error instead of asserting the specific error text? The extra mile of asserting the specific text is noble but we also tend to avoid such checks since they are brittle to change. (when we run across cases like this in the code base that we think are really important to check against, we'll make sentinel errors and use `errors.Is` / `assert.ErrorIs` during checks).

Reply:
Sure. Do you want a sentinel error type in this case, or just the removal of expectedError string?

Reply:
I can't think of a specific reason why the caller would want to check this case specifically, so my vote would be on the simpler side --replace the string assertion with an existence check (assert we expect an error in this case, but comparing the error isn't necessary).